code/modules/overmap/ftl_shunt/computer.dm

```diff
@@ -0,0 +1,150 @@
+/obj/machinery/computer/ship/ftl
+	name = "superluminal shunt control console"
+	icon_keyboard = "teleport_key"
+	icon_screen = "teleport"
+	light_color = "#77fff8"
+	var/obj/machinery/ftl_shunt/core/linked_core
+	var/cost = 0
+
+/obj/machinery/computer/ship/ftl/Initialize()
+	. = ..()
+	find_core()
+
+/obj/machinery/computer/ship/ftl/Destroy()
+	. = ..()
+	if(linked_core)
+		linked_core.ftl_computer = null
+		linked_core = null
+
+/obj/machinery/computer/ship/ftl/proc/recalc_cost()
+	var/jump_dist = get_dist(linked, locate(linked_core.shunt_x, linked_core.shunt_y, GLOB.using_map.overmap_z))
+	var/jump_cost = ((linked.vessel_mass * JOULES_PER_TON) / 1000) * jump_dist
+	return jump_cost
+
+/obj/machinery/computer/ship/ftl/proc/find_core()
+	if(!linked)
+		return
+
+	if(linked_core)
+		linked_core.ftl_computer = null
+		linked_core = null
+
+	for(var/obj/machinery/ftl_shunt/core/C in SSmachines.machinery)
+		if(C.z in linked.map_z)
+			linked_core = C
+			C.ftl_computer = src
+
+/obj/machinery/computer/ship/ftl/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+	if(!linked)
+		display_reconnect_dialog(user, "superluminal shunt management system")
+		return
+
+	var/data[0]
+
+	if(!linked_core)
+		find_core()
+		if(!linked_core)
+			to_chat(user, SPAN_WARNING("Unable to establish connection to superluminal shunt."))
+			return
+	recalc_cost()
```

Comment:
Seems a bit strange to calc this during the UI update - you might be doing multiple updates per second, etc. Could the cost be calculated when you find your core maybe?

Reply:
Cost is calculated during UI update because doing it whenever the destination x/y changed was a little clunky. Jump cost is a factor of mass * distance so it is dynamic depending on how far you're jumping, so it requires updating.

Reply:
Fair enough.